### PR TITLE
Added gcs-media plugin

### PR DESCRIPTION
**Pull request by tmatsuo**
Now the unit tests are passing!

Remaining TODOs:
- [x] Add readme.txt including how to install google/cloud
- [x] Add google/cloud and install the GCS stream wrapper in the test bootstrap
- [x] Set environment variable on travis (`TEST_BUCKET`, `GOOGLE_APPLICATION_CREDENTIALS`, and `GOOGLE_CREDENTIALS_BASE64`)

This plugin depends on https://github.com/GoogleCloudPlatform/google-cloud-php/issues/23

### CODE REVIEW

File: gcs-media-plugin/tests/test-gcs.php [hunk 1]
```diff
@@ -0,0 +1,187 @@
...
+        // We have the option set to true (1).
+        \Google\Cloud\Storage\WordPress\register_settings();
+        $ssl = get_option(Uploads::USE_HTTPS_OPTION);
+        $this->assertEquals(1, $ssl);
+    }
+
+    /**
+     * A test for filter_delete_file.
+     */
+    public function test_filter_delete_file()
```
Comment (chingor13) [inline gcs-media-plugin/tests/test-gcs.php]: This test really doesn't seem to test much, although it looks like it's hard to make sure `unlink` was called.

File: gcs-media-plugin/tests/test-gcs.php [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+            'path' => '/tmp/uploads',
+            'subdir' => '/2016/11',
+            'url' => 'https://example.com/wp-content/2016/11/uploaded.jpg',
+            'basedir' => '/tmp/uploads',
+            'baseurl' => 'https://example.com/wp-content/2016/11/',
+            'error' => false
+        );
+        \update_option(Uploads::BUCKET_OPTION, $testBucket);
+        $values = Uploads::filter_upload_dir($values);
+        var_dump($values);
```
Comment (chingor13) [inline gcs-media-plugin/tests/test-gcs.php]: Do we want to add some assertions here to see if it's doing what we expect?

File: gcs-media-plugin/gcs.php [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+Description: A plugin for uploading media files to Google Cloud Storage
+Version:     0.1
+Author:      Google Inc
+Author URI:  http://cloud.google.com/
+License:     GPL2
+License URI: https://www.gnu.org/licenses/gpl-2.0.html
+Text Domain: gcp
+Domain Path: /languages
+
+Copyright 2016 Google Inc.
```
Comment (chingor13) [inline gcs-media-plugin/gcs.php]: 2017

File: gcs-media-plugin/Uploads/Uploads.php [hunk 4]
```diff
@@ -0,0 +1,205 @@
+<?php
+/**
+ * Uploading functionality with Google Cloud Storage
+ *
+ * Hijacks the uploading functionality in WordPress to use Google Cloud Storage
+ * for the media library.
+ *
+ * Copyright 2016 Google Inc.
```
Comment (chingor13) [inline gcs-media-plugin/Uploads/Uploads.php]: 2017

File: gcs-media-plugin/tests/test-gcs.php (hunk 1 above)
Comment (tmatsuo) [inline gcs-media-plugin/tests/test-gcs.php]: Yeah filter_delete_file should be removed in the future.  I don't think it's worthwhile to invest here.

File: gcs-media-plugin/tests/test-gcs.php (hunk 2 above)
Comment (tmatsuo) [inline gcs-media-plugin/tests/test-gcs.php]: Done

File: gcs-media-plugin/gcs.php (hunk 3 above)
Comment (tmatsuo) [inline gcs-media-plugin/gcs.php]: Done

File: gcs-media-plugin/Uploads/Uploads.php (hunk 4 above)
Comment (tmatsuo) [inline gcs-media-plugin/Uploads/Uploads.php]: Done

**Comment by tmatsuo** [comment]
@chingor13 Thanks, PTAL!

**Pull request merged**